common/main/java/com/couchbase/lite/internal/replicator/AbstractCBLWebSocket.java

```diff
@@ -653,23 +650,39 @@ private Request newRequest() {
         return builder.build();
     }
 
-    @Nullable
-    private Authenticator getBasicAuthenticator() {
-        if (options == null) { return null; }
+    private void getBasicAuthenticator(@NonNull OkHttpClient.Builder builder) {
+        if (options == null) { return; }
 
         final Object obj = options.get(C4Replicator.REPLICATOR_OPTION_AUTHENTICATION);
-        if (!(obj instanceof Map)) { return null; }
+        if (!(obj instanceof Map)) { return; }
         final Map<?, ?> auth = (Map<?, ?>) obj;
 
         final Object authType = auth.get(C4Replicator.REPLICATOR_AUTH_TYPE);
-        if (!C4Replicator.AUTH_TYPE_BASIC.equals(authType)) { return null; }
+        if (!C4Replicator.AUTH_TYPE_BASIC.equals(authType)) { return; }
 
         final Object username = auth.get(C4Replicator.REPLICATOR_AUTH_USER_NAME);
-        if (!(username instanceof String)) { return null; }
+        if (!(username instanceof String)) { return; }
         final Object password = auth.get(C4Replicator.REPLICATOR_AUTH_PASSWORD);
-        if (!(password instanceof String)) { return null; }
-
-        return (route, response) -> authenticate(response, (String) username, (String) password);
+        if (!(password instanceof String)) { return; }
+
+        final String credentials = Credentials.basic((String) username, (String) password);
+
+        builder.authenticator((route, response) -> authenticate(response, credentials));
```

Comment:
Is this still needed as the auth is already included in the header? I understand that the challenge will not be called again if the auth fails.

Reply:
I am absolutely not certain....  It seems correct to respond to a challenge, if I ever get it.  If I don't ever get it... 🤷🏻